tests/io/test_inifield.py

```diff
@@ -0,0 +1,165 @@
+import inspect
+from contextlib import nullcontext as does_not_raise
+from pathlib import Path
+from typing import Any, List, Union
+
+import pytest
+from pydantic.error_wrappers import ValidationError
+from pydantic.types import FilePath
+
+from hydrolib.core.io.ini.parser import Parser, ParserConfig
+from hydrolib.core.io.inifield.models import (
+    AveragingType,
+    DataFileType,
+    IniFieldGeneral,
+    IniFieldModel,
+    InitialField,
+    InterpolationMethod,
+    LocationType,
+    Operand,
+    ParameterField,
+)
+
+from ..utils import WrapperTest, invalid_test_data_dir, test_data_dir
+
+
+class TestIniField:
+    def _get_datafiletype_cases() -> List:
```

Comment:
There are a couple of SC bugs regarding these functions. Please add `self` to the function signature.

Reply:
Solved by making static variables, because the functions as true object functions could not be called in `@pytest.mark.parametrize`. I also could not get `@staticmethod` to work, therefore, as static class variables. Now alle these tests work well.